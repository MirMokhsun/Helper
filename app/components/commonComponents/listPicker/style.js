import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        width: 'auto',
        minHeight: 66,
        borderBottomWidth: 1,
        borderBottomColor: '#D3D9E7',
        marginHorizontal: 15,
        flexDirection: 'row',
        alignItems: 'center'
    },
    text: {
        fontFamily: 'Roboto-Light',
        fontSize: 16,
        color: '#000756',
        lineHeight: 21,
        flex: 1,
        paddingTop: 15,
        paddingBottom: 15,
    },
    circleWrapper: {
        width: '20%',
        height: 65,
        flexDirection: 'row-reverse',
        alignItems: 'center'
    },
    circle: {
        width: 24,
        height: 24,
        borderRadius: 12,
        backgroundColor: '#EBECEE',
        alignItems: 'center',
        justifyContent: 'center',
    },
    circleChecked: {
        width: 14,
        height: 14,
        borderRadius: 7,
        backgroundColor: '#1459D2',
    }
});