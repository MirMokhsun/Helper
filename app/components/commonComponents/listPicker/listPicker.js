import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { styles } from './style';

export class ListPicker extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        const { showCirle, text, checked, onPress, id, currentTextAlign } = this.props;
        return (
            <TouchableOpacity style={styles.container} onPress={onPress} testID={`ListPicker${id}ID`} accessibilityLabel={`ListPicker${id}ID`}>
                <Text style={currentTextAlign ? [styles.text, { textAlign: currentTextAlign }] : [styles.text, { textAlign: 'left' }]}>
                    {text}
                </Text>
                {showCirle ? (
                    <View style={styles.circleWrapper}>
                        <View style={styles.circle} >
                            {checked ? <View style={styles.circleChecked} /> : null}
                        </View>
                    </View>
                ) : null}
            </TouchableOpacity>
        );
    }
}   
