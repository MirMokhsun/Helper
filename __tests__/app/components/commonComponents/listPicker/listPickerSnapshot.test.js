import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ListPicker } from '../../../../../app/components/commonComponents/listPicker/listPicker';

it('create snapshot ListPicker', () => {
    const renderer = new ShallowRenderer()
    let listPicker = renderer.render(<ListPicker />);
    expect(listPicker).toMatchSnapshot();
});