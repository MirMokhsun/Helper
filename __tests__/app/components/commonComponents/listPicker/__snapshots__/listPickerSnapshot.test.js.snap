// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ListPicker 1`] = `
<TouchableOpacity
  accessibilityLabel="ListPickerundefinedID"
  activeOpacity={0.2}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#D3D9E7",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginHorizontal": 15,
      "minHeight": 66,
      "width": "auto",
    }
  }
  testID="ListPickerundefinedID"
>
  <Text
    style={
      Array [
        Object {
          "color": "#000756",
          "flex": 1,
          "fontFamily": "Roboto-Light",
          "fontSize": 16,
          "lineHeight": 21,
          "paddingBottom": 15,
          "paddingTop": 15,
        },
        Object {
          "textAlign": "left",
        },
      ]
    }
  />
</TouchableOpacity>
`;
